src/library.js

```diff
@@ -629,27 +673,31 @@ const createPilingJs = (rootElement, initOptions = {}) => {
 
     if (pileInstances) {
       pileInstances.forEach((pile, id) => {
-        let i;
-        let j;
-        if (items[id].position) {
-          [i, j] = items[id].position;
-        } else {
-          const getCellPosition = orderer(layout.numColumns);
-          [i, j] = getCellPosition(id);
-        }
+        let x = null;
+        let y = null;
 
-        layout.numRows = j + 1;
+        [x, y] = arrangePile(pile);
 
-        const [x, y] = layout.ijToXy(
-          i,
-          j,
-          pile.graphics.width,
-          pile.graphics.height
-        );
+        if (x === null) {
+          const getCellPosition = orderer(layout.numColumns);
+          const [i, j] = getCellPosition(id);
 
-        renderedItems.get(id).originalPosition = [x, y];
+          layout.numRows = j + 1;
 
-        movingPiles.push({ id, x, y });
+          [x, y] = layout.ijToXy(
+            i,
+            j,
+            pile.graphics.width,
+            pile.graphics.height
+          );
+          movingPiles.push({ id, x, y });
```

Comment:
I think this condition should be part of `getPilePosition`. This function should only have to deal with calling `getPilePosition()` for every pile, dispatching the `movePiles` action, and updating `layout.numRows`

Reply:
Hmm, this condition is when `type` is `undefined` or `null`, the piles are initially positioned once and are not automatically moved, so I didn't put this condition in `getPilePosition()`.